tests/common/txn_test.go

```diff
@@ -0,0 +1,204 @@
+			ifSucess: []string{"get key1", "get key2", `put "key \"with\" space" "value \x23"`},
+			ifFail:   []string{`put key1 "fail"`, `put key2 "fail"`},
+			results:  []string{"SUCCESS", "key1", "value1", "key2", "value2", "OK"},
+		},
+		{
+			compare:  []string{`version("key \"with\" space") = "1"`},
+			ifSucess: []string{`get "key \"with\" space"`},
+			results:  []string{"SUCCESS", `key "with" space`, "value \x23"},
+		},
+	}
+	setup := func(cc framework.Client) {
+		if err := cc.Put("key1", "value1", config.PutOptions{}); err != nil {
+			t.Fatalf("could not create key:%s, value:%s", "key1", "value1")
+		}
+		if err := cc.Put("key2", "value2", config.PutOptions{}); err != nil {
+			t.Fatalf("could not create key:%s, value:%s", "key2", "value2")
+		}
+	}
+	testTxn(t, tcs, setup, reqs)
+}
+
+func TestTxnFail(t *testing.T) {
+	tcs := []config.NamedClusterConfig{
+		{
+			Name:   "NoTLS",
+			Config: config.ClusterConfig{ClusterSize: 1},
+		},
+		{
+			Name:   "PeerTLS",
+			Config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.ManualTLS},
+		},
+		{
+			Name:   "PeerAutoTLS",
+			Config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.AutoTLS},
+		},
+		{
+			Name:   "ClientTLS",
+			Config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.ManualTLS},
+		},
+		{
+			Name:   "ClientAutoTLS",
+			Config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.AutoTLS},
+		},
+	}
+	reqs := []txnReq{
+		{
+			compare:  []string{`version("key") < "0"`},
+			ifSucess: []string{`put key "success"`},
+			ifFail:   []string{`put key "fail"`},
+			results:  []string{"FAILURE", "OK"},
+		},
+		{
+			compare:  []string{`value("key1") != "value1"`},
+			ifSucess: []string{`put key1 "success"`},
+			ifFail:   []string{`put key1 "fail"`},
+			results:  []string{"FAILURE", "OK"},
+		},
+	}
+	setup := func(cc framework.Client) {
+		if err := cc.Put("key1", "value1", config.PutOptions{}); err != nil {
+			t.Fatalf("could not create key:%s, value:%s", "key1", "value1")
+		}
+	}
+	testTxn(t, tcs, setup, reqs)
+}
+
+func getRespValues(r *clientv3.TxnResponse) []string {
+	ss := []string{}
+	if r.Succeeded {
+		ss = append(ss, "SUCCESS")
+	} else {
+		ss = append(ss, "FAILURE")
+	}
+	for _, resp := range r.Responses {
+		switch v := resp.Response.(type) {
+		case *pb.ResponseOp_ResponseDeleteRange:
+			r := (clientv3.DeleteResponse)(*v.ResponseDeleteRange)
+			ss = append(ss, fmt.Sprintf("%d", r.Deleted))
+		case *pb.ResponseOp_ResponsePut:
+			r := (clientv3.PutResponse)(*v.ResponsePut)
+			ss = append(ss, "OK")
+			if r.PrevKv != nil {
+				ss = append(ss, string(r.PrevKv.Key), string(r.PrevKv.Value))
+			}
+		case *pb.ResponseOp_ResponseRange:
+			r := (clientv3.GetResponse)(*v.ResponseRange)
+			for _, kv := range r.Kvs {
+				ss = append(ss, string(kv.Key), string(kv.Value))
+			}
+		default:
+			ss = append(ss, fmt.Sprintf("\"Unknown\" : %q\n", fmt.Sprintf("%+v", v)))
+		}
+	}
+	return ss
+}
+
+func Test_AddTxnResponse(t *testing.T) {
```

Comment:
This looks like unitest for `tests/framework/common` and not integration/e2e tests for `tests/common`. Please move.

Reply:
yes it is.

Reply:
updated